sway-core/src/semantic_analysis/ast_node/expression/intrinsic_function.rs

```diff
@@ -92,10 +92,117 @@ impl ty::TyIntrinsicFunctionKind {
             Intrinsic::ContractRet => {
                 type_check_contract_ret(handler, ctx, kind, arguments, type_arguments, span)
             }
+            Intrinsic::EncodeBufferEmpty => {
+                type_check_encode_buffer_empty(ctx, kind, arguments, type_arguments, span)
+            }
+            Intrinsic::EncodeBufferAppend => {
+                type_check_encode_append(handler, ctx, kind, arguments, type_arguments, span)
+            }
+            Intrinsic::EncodeBufferAsRawSlice => {
+                type_check_encode_as_raw_slice(handler, ctx, kind, arguments, type_arguments, span)
+            }
         }
     }
 }
 
+fn type_check_encode_as_raw_slice(
+    handler: &Handler,
+    mut ctx: TypeCheckContext,
+    kind: sway_ast::Intrinsic,
+    arguments: &[Expression],
+    _type_arguments: &[TypeArgument],
+    span: Span,
+) -> Result<(ty::TyIntrinsicFunctionKind, TypeId), ErrorEmitted> {
+    let type_engine = ctx.engines.te();
+    let engines = ctx.engines();
+
+    let buffer_expr = {
+        let ctx = ctx
+            .by_ref()
+            .with_help_text("")
+            .with_type_annotation(type_engine.insert(engines, TypeInfo::Unknown, None));
```

Comment:
Shouldn't we have `Buffer` here as type annotation? If we call e.g. `__encode_buffer_as_raw_slice(true)` this will pass type check phase and the compiler will, expectantly, panic in both const context and in non-const context.

Reply:
Done